src/pcapi/core/users/api.py

```diff
@@ -283,3 +289,65 @@ def user_expenses(user: User):
         )
 
     return limits
+
+
+def create_pro_user(body: ProUserCreationBodyModel) -> User:
+    objects_to_save = []
+
+    new_user = User(from_dict=body.dict(by_alias=True))
+    new_user.hasAllowedRecommendations = body.contact_ok
+
+    existing_offerer = Offerer.query.filter_by(siren=body.siren).first()
+
+    if existing_offerer:
+        user_offerer = _generate_user_offerer_when_existing_offerer(new_user, existing_offerer)
+        offerer = existing_offerer
+    else:
+        offerer = _generate_offerer(body.dict(by_alias=True))
+        user_offerer = offerer.give_rights(new_user, RightsType.editor)
+        digital_venue = create_digital_venue(offerer)
+        objects_to_save.extend([digital_venue, offerer])
+    objects_to_save.append(user_offerer)
+    new_user.isBeneficiary = False
+    new_user.isAdmin = False
+    new_user.needsToFillCulturalSurvey = False
+    new_user = _set_offerer_departement_code(new_user, offerer)
+
+    new_user.generate_validation_token()
+    objects_to_save.append(new_user)
+
+    repository.save(*objects_to_save)
```

Comment:
On append seulement l'utilisateur à ce tableau. A priori, il ne sert à rien sauf si j'ai loupé quelque chose ? 🤔
```suggestion
    repository.save(new_user)
```

Reply:
Non, il y a aussi digital_venue et offerer dans un else + user_offerer en dehors